docs/DeveloperGuide.md

```diff
@@ -287,16 +287,29 @@ _{Explain here how the data archiving feature will be implemented}_
+| ***      | NGO administrator | search for contacts by name, role                                         | I can quickly find relevant personnel                                                                   |
+| ***      | NGO administrator | create groups of contacts (e.g., volunteers, donors, partners)            | I can manage communications more efficiently                                                            |
+| ***      | NGO administrator | add a new volunteer's contact information                                 | I can add volunteers to my organisation                                                                 |
+| ***      | NGO administrator | set a volunteer's participation hours                                     | I can see volunteer's participation hours                                                               |
+| ***      | NGO administrator | edit a volunteer's information                                            | I can keep volunteer records up to date                                                                 |
+| ***      | NGO administrator | delete a volunteer's contact information                                  | I can remove irrelevant or erroneous entries                                                            |
+| **       | NGO administrator | sort contacts by alphabetical or insertion order                          | that I can easily locate and manage contacts efficiently.                                               |
+| **       | NGO administrator | see a dashboard of recent activity and upcoming tasks                     | I can have an overview of key engagements at a glance.                                                  |
+| **       | NGO administrator | merge duplicate contact entries                                           | I can maintain a clean and organized address book                                                       |
+| **       | NGO administrator | archive contacts who are no longer active                                 | I can keep the active contact list clean without deleting important data                                |
+| **       | Donor             | view donation history                                                     | I can easily track my donations overtime                                                                |
+| **       | Donor             | view donation total                                                       | I can see how much I have donated in total                                                              |
+| **       | Volunteer         | update my contact information directly within the system                  | I can ensure my details are always accurate for the NGO.                                                |
```

Comment:
I think this could violate the single user constraint? Maybe we should only have NGO administrator as our user